bundlegen/src/main/java/com/citrix/microapps/bundlegen/bundles/BundlesLoader.java

```diff
@@ -98,6 +132,28 @@ public Bundle loadBundle(FsBundle bundle) {
                 .collect(Collectors.toList());
     }
 
+    static List<ValidationException> checkLocalizations(FsBundle bundle) {
+        List<ValidationException> issues = new ArrayList<>();
+        loadAndValidateTemplateFile(issues, bundle)
+                .ifPresent(template ->
+                        bundle.getFiles().stream()
+                                .filter(path -> BUNDLE_ALLOWED_TRANSLATIONS.contains(path))
+                                .forEach(translationPath ->
+                                        loadTranslationFile(issues, bundle.getPath().resolve(translationPath))
+                                                .filter(modelTranslation ->
+                                                        !Objects.equals(
+                                                                new TranslationValidator(modelTranslation).checksum(),
+                                                                template.getTranslationChecksum()))
+                                                .ifPresent(modelTranslation ->
+                                                        issues.add(new ValidationException(
+                                                                String.format("Bundle has changed translation keys %s",
+                                                                        translationPath.getFileName())))
```

Comment:
Please replace side effects in `ifPresent()` + `forEach()` by `map()` + `flatMap()` + `collect()`. Consider to split the code to multiple methods, this is difficult to read.

Reply:
Replaced with `map()` and `.filter(Optional::isPresent)`